src/fhirtypes/common.ts

```diff
@@ -41,22 +41,44 @@ export function setPropertyOnInstance(
       const key =
         pathPart.primitive && i < pathParts.length - 1 ? `_${pathPart.base}` : pathPart.base;
       // If this part of the path indexes into an array, the index will be the last bracket
-      const index = getArrayIndex(pathPart);
+      let index = getArrayIndex(pathPart);
+      let sliceName: string;
       if (index != null) {
         // If the array doesn't exist, create it
         if (current[key] == null) current[key] = [];
+        sliceName = getSliceName(pathPart);
+        if (sliceName) {
+          const sliceIndices: number[] = [];
+          // Find the indices where slices are placed
+          current[key].forEach((el: any, i: number) => {
+            if (el?._sliceName === sliceName) {
+              sliceIndices.push(i);
+            }
+          });
+          // Convert the index in terms of the slice to the corresponding index in the overall array
+          if (index >= sliceIndices.length) {
+            index = index - sliceIndices.length + current[key].length;
```

Comment:
I'm having a little trouble following... so this may or may not be a good question, but let's assume we have:
```
foo: [
  { _sliceName: 'a', value: 'a0' },
  { _sliceName: 'a', value: 'a1' },
  { _sliceName: 'b', value: 'b0' },
  { value: 'sliceless0' }
]
```
Based on the logic above, would `foo[a][2].value = "a2"` end up overwriting the first element of slice `b`?  Don't you need to insert an element in the array between the last slice a and the first slice b?

Again, might not be a valid question.  Just want to be sure.